lib/pymedphys/_trf/decode/header.py

```diff
@@ -38,34 +52,62 @@ def determine_header_length(trf_contents: bytes) -> int:
         TRF file.
     """
 
-    column_end = b"\t\xdc\x00\xe8\t\xdc\x00\xe9\t\xdc\x00\xea\t\xdc\x00\xeb\t\xdc\x00"
-    header_length = trf_contents.index(column_end) + len(column_end)
+    # Date regex match (any 25 characters \x00-\x19) with pattern (\d\d/\d\d/\d\d \d\d:\d\d:\d\d Z)
+    # UTC offset regex (any 25 characters \x00-\x19) with pattern ([\+|\-]\d\d:\d\d)
+    # Treatment name regex match (any 50 characters \x00-\x32) with pattern ([\x20-\x7F]*) according to ASCII table encoding
+    # Machine number (any 10 characters \x00-\x10) with pattern ([\x20-\x7F]*) according to ASCII table encoding
+    # Treatment Record related header
+
+    regex_trf = (
+        rb"[\x00-\x19]"
+        + rb"(\d\d[/|-]\d\d[/|-]\d\d \d\d:\d\d:\d\d Z)"
+        + rb"[\x00-\x19]"
+        + rb"([\+|\-]\d\d:\d\d)"
+        + rb"[\x00-\x32]"
+        + rb"([\x20-\x7F]*)"
+        + rb"[\x00-\x10]"
+        + rb"([\x20-\x7F]*)"
+        + rb"([\x00-\xFFF]*)"
+    )
+
+    # The first 4 groups in the regex match are dynamically allocated as
+    # depend on the date, timezone, field and machine
+    # The field is likely to change the number of bytes.
+
+    start_header_length = 0
+    match = re.match(regex_trf, trf_contents)
 
-    # test = trf_contents.split(b"\t")
-    # row_skips = 6
-    # i = next(i for i, item in enumerate(test[row_skips::]) if len(item) > 3) + row_skips
-    # header_length_old_method = len(b"\t".join(test[0:i])) + 3
+    if match:
+        groups = match.groups()
+        start_header_length = match.span(4)[1]
 
-    # if header_length_old_method != header_length:
-    #     raise ValueError("Inconsistent header length determination")
+    else:
+        raise ValueError("Inconsistent header length determination")
```

Comment:
This error message should probably change to represent the new issue type "Unexpected header content found", or something like that.

Reply:
Yes. I'll amend accordingly.